### PR TITLE
Enable in-transit encryption for external mode deployment

### PR DESCRIPTION
**Pull request by vavuthu**
Signed-off-by: vavuthu [vavuthu@redhat.com](mailto:vavuthu@redhat.com)

### CODE REVIEW

**Comment by vavuthu** [comment]
verified here: https://url.corp.redhat.com/bf89d00

File: ocs_ci/deployment/helpers/external_cluster_helpers.py [hunk 1]
```diff
@@ -411,6 +411,19 @@ def remove_rbd_images(self, images, pool):
             cmd = f"rbd rm {each_image} -p {pool}"
             self.rhcs_conn.exec_cmd(cmd)
 
+    def enable_secure_connection_mode(self):
+        """
+        Enables secure connection mode for RHCS cluster
+        """
+        logger.debug("Enabling secure connection mode for external RHCS cluster")
+        cmds = (
+            "ceph config set global ms_client_mode secure;"
```
Comment (petr-balogh) [inline ocs_ci/deployment/helpers/external_cluster_helpers.py:420]: Do we need to disable it in the teardown or doesn't matter?

File: ocs_ci/deployment/deployment.py [hunk 2]
```diff
@@ -1328,12 +1328,23 @@ def deploy_with_external_mode(self):
 
         cluster_data = templating.load_yaml(constants.EXTERNAL_STORAGE_CLUSTER_YAML)
         cluster_data["metadata"]["name"] = config.ENV_DATA["storage_cluster_name"]
+
+        # Enable in-transit encryption.
+        if config.ENV_DATA.get("in_transit_encryption"):
+            cluster_data["spec"]["network"] = {
```
[the remark this answers is not in the document]
Comment (vavuthu) [reply ocs_ci/deployment/deployment.py:1334]: for external cluster, we don't have any network attribute so far.

File: ocs_ci/deployment/helpers/external_cluster_helpers.py (hunk 1 above)
Comment (vavuthu) [reply ocs_ci/deployment/helpers/external_cluster_helpers.py:420]: Its better to disable during teardown. Tested on UMB cluster and even though if we ran the same commands twice, it doesn't throw any error. I will do it as a followup PR.

**Comment by vavuthu** [comment]
acceptance testing: https://url.corp.redhat.com/2200eb9

**Comment by vavuthu** [comment]
BUILD ID: 4.13.0-214 RUN ID: 1686048740
```
Summary

28 tests ran in 3205.84 seconds.
28 passed, 6 skipped, 0 failed, 0 errors, 0 expected failures, 0 unexpected passes
```

**Review by shylesh — APPROVED** [review]
+1

**Pull request merged**